python/paddle/fluid/layers/tensor.py

```diff
@@ -332,6 +333,8 @@ def concat(input, axis=0, name=None):
         check_variable_and_dtype(
             x, 'input[' + str(id) + ']',
             ['float16', 'float32', 'float64', 'int32', 'int64'], 'concat')
+        if convert_dtype(x.dtype) != convert_dtype(input[0].dtype):
+            raise TypeError("All Tensor in the input must has same data type.")
```

Comment:
```suggestion
            raise TypeError("All Tensors in the input must have same data type.")
```

同样一句话出现在几个位置，每次都不相同，请仔细检查一下是否还有其他问题

Reply:
Done